subworkflows/ebi-metagenomics/fasta_domainannotation/main.nf

```diff
@@ -0,0 +1,45 @@
+include { BLAST_MAKEBLASTDB } from '../../../modules/ebi-metagenomics/blast/makeblastdb/main'
+include { BLAST_BLASTP      } from '../../../modules/ebi-metagenomics/blast/blastp/main'
+include { DIAMOND_MAKEDB    } from '../../../modules/ebi-metagenomics/diamond/makedb/main'
+include { DIAMOND_BLASTP    } from '../../../modules/ebi-metagenomics/diamond/blastp/main'
+include { INTERPROSCAN      } from '../../../modules/ebi-metagenomics/interproscan/main'
+
+workflow FASTA_DOMAINANNOTATION {
+
+    take:
+    ch_fasta       // channel: [ val(meta), path(fasta) ]
+    ch_blast_fasta // channel: /path/to/reference/fasta for blast
+    val_blast_mode // value: blast or diamond
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    if (val_blast_mode == "blast") {
+        println("blast mode")
+        BLAST_MAKEBLASTDB ( ch_blast_fasta )
+        ch_versions = ch_versions.mix(BLAST_MAKEBLASTDB.out.versions)
+        BLAST_BLASTP ( ch_fasta, BLAST_MAKEBLASTDB.out.db, 'tsv' )
```

Comment:
Have you consider having for steps in the same process? The MAKEBLAST and BLASTP?

By doing so you can lower a but needless data linking, in a cloud setting it would introduce quite the overhead as it would have to copy the files over.